dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/TrackerImportController.java

```diff
@@ -114,25 +112,15 @@ public TrackerImportReport syncPostJsonTracker(
 
         TrackerImportParams trackerImportParams = buildTrackerImportParams( request, currentUser, trackerBundleParams );
         TrackerImportReport trackerImportReport = trackerImportService.importTracker( trackerImportParams );
-
-        return trackerImportService.buildImportReport( trackerImportReport, trackerBundleReportMode );
-
-    }
-
-    private TrackerBundleReportMode getTrackerBundleReportMode( String reportMode )
-    {
-
-        TrackerBundleReportMode trackerBundleReportMode;
-        try
-        {
-            trackerBundleReportMode = TrackerBundleReportMode.valueOf( reportMode.toUpperCase() );
-        }
-        catch ( IllegalArgumentException e )
-        {
-            throw new HttpClientErrorException( HttpStatus.BAD_REQUEST,
-                "Value " + reportMode + " is not a valid report mode" );
-        }
-        return trackerBundleReportMode;
+        TrackerImportReport trackerImportReportResponse = trackerImportService.buildImportReport( trackerImportReport,
+            trackerBundleReportMode );
+
+        return trackerImportReportResponse.getStatus() == TrackerStatus.ERROR ? new ResponseEntity<>(
+            trackerImportReportResponse,
+            HttpStatus.CONFLICT )
+            : new ResponseEntity<>(
+                trackerImportReportResponse,
+                HttpStatus.OK );
```

Comment:
Just as a suggestion, might be easier to read:
feel free to decide to apply or not

```suggestion
        ResponseEntity.BodyBuilder builder = 
                trackerImportReportResponse.getStatus() == TrackerStatus.ERROR ?
                        ResponseEntity.status(HttpStatus.CONFLICT) :
                        ResponseEntity.ok();
        
        return builder.body(trackerImportReportResponse);
```

Reply:
yes good suggestion i didn't remember about the builder